### PR TITLE
Add connect context arg to Connection::connect()

### PR DESCRIPTION
**Pull request by trowski**
Addresses #81.

### CODE REVIEW

**Comment by trowski** [comment]
@prolic @kelunik Please review.

**Comment by prolic** [comment]
Looks good to me. I didn't check for postgres, maybe it's the same problem there? Can you also add a test?

File: src/TimeoutConnector.php [hunk 1]
```diff
@@ -30,14 +29,6 @@ public function __construct(int $timeout = self::DEFAULT_TIMEOUT)
      */
     public function connect(SqlConnectionConfig $config): Promise
     {
-        if (!$config instanceof ConnectionConfig) {
-            throw new \TypeError(\sprintf("Must provide an instance of %s to MySQL connectors", ConnectionConfig::class));
-        }
```
Comment (kelunik) [inline src/TimeoutConnector.php]: This shouldn't have been dropped I guess?

Comment (trowski) [reply src/TimeoutConnector.php]: Yeah, I was a little overzealous with the delete key.

**Comment by trowski** [comment]
@kelunik As far as I can tell the previous way worked too, though this way is better since a timeout >10 sec would have been overridden by the socket context.

@prolic I don't think this is a problem in postgres since that library uses loop watchers for connection due to how the extensions work.

**Comment by bwoebi** [comment]
Shouldn't TimeoutConnector gain that new context arg as well?

**Comment by bwoebi** [comment]
I meant in particular because of pool, which is using TimeoutConnector internally

**Comment by kelunik** [comment]
Any reason why this has not been merged?

**Pull request merged**